app_check/src/desktop/app_check_desktop.cc

```diff
@@ -57,9 +102,21 @@ Future<AppCheckToken> AppCheckInternal::GetAppCheckTokenLastResult() {
       future()->LastResult(kAppCheckFnGetAppCheckToken));
 }
 
-void AppCheckInternal::AddAppCheckListener(AppCheckListener* listener) {}
+void AppCheckInternal::AddAppCheckListener(AppCheckListener* listener) {
+  if (listener) {
+    token_listeners_.push_back(listener);
+
+    if (HasValidCacheToken()) {
```

Comment:
This is probably correct, though iOS does behave differently.

It seems that android does notify immediately if a valid token already exists, so this is consistent with android.
For iOS, token-changed listener is based on notifications being sent and adding a listener will not retroactively receive old notifications.

So this can probably stay as is.

Reply:
Yeah, I copied this logic from the Android one. The joys of trying to resolve two different implementations. :)

Reply:
I put a comment calling out the difference.